src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs

```diff
@@ -3625,6 +3625,35 @@ public async Task CanPassDerivedParameterToStreamHubMethod(string method)
             }
         }
 
+        [Fact]
+        public async Task ClientsCallerPropertyCanBeUsedOutsideOfHub()
+        {
+            CallerService callerService = new CallerService();
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(services =>
+            {
+                services.AddSingleton(callerService);
+            });
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<CallerServiceHub>>();
+
+            using (StartVerifiableLog())
+            {
+                using (var client = new TestClient())
+                {
+                    var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+
+                    // Wait for a connection, or for the endpoint to fail.
+                    await client.Connected.OrThrowIfOtherFails(connectionHandlerTask).OrTimeout();
```

Comment:
Does TestClient.ConnectAsync() normally complete before Hub.OnConnectedAsync() does?

Reply:
Probably, whats the hidden question here?

Reply:
Intuitively I would expect TestClient.ConnectAsync() *after* Hub.OnConnectedAsync(). It's weird if that's not the case, though that's really not the point of this PR which is why I approved it anyway.

Setting this Connected TCS seems like a lot of work to for nothing, so I'm guessing TestClient.ConnectAsync() actually does normally complete before Hub.OnConnectedAsync(). Is that right? Is that also true for non-test clients?

Reply:
>I'm guessing TestClient.ConnectAsync() actually does normally complete before Hub.OnConnectedAsync(). Is that right? Is that also true for non-test clients?

That's correct, `ConnectAsync` waits for the handshake response and then returns for both test and non-test clients.

I've thought about waiting for `Hub.OnConnectedAsync` before and would love to discuss it further, but not on this PR 😉